lib/THC/THCTensorMath.cuh

```diff
@@ -23,4 +23,31 @@ __global__ void THCTensor_copyToDiagonal(T* a, T* b, ptrdiff_t start, ptrdiff_t
   }
 }
 
+#define CAT_ARRAY_KERNEL_BATCH_SIZE 2
```

Comment:
(and don't use TensorInfo perhaps, or you'd have to add the size of the size/stride array as a template parameter, and/or split TensorInfo into a base which is just size only)